src/basilisp/lang/obj.py

```diff
@@ -76,17 +77,46 @@ def map_lrepr(
     with the start and end string supplied. The entries argument must be a
     callable which will produce tuples of key-value pairs.
 
+    If the keyword argument print_namespace_maps is True and all keys
+    are symbols or keywords and they share the same namespace, then
+    print the namespace of the keys at the beginning of the map
+    instead of beside the keys.
+
     The keyword arguments will be passed along to lrepr for the sequence
-    elements."""
+    elements.
+
+    """
+    from basilisp.lang import keyword as kw  # pylint: disable=cyclic-import
+    from basilisp.lang import symbol as sym  # pylint: disable=cyclic-import
+
     print_level = kwargs["print_level"]
     if isinstance(print_level, int) and print_level < 1:
         return SURPASSED_PRINT_LEVEL
 
     kwargs = _process_kwargs(**kwargs)
 
+    def check_same_ns():
+        nses = set()
+        for k, _ in entries():
+            if isinstance(k, (kw.Keyword, sym.Symbol)):
+                nses.add(k.ns)
+                if len(nses) > 1:
+                    break
+        return next(iter(nses)) if len(nses) == 1 else None
+
+    print_namespace_maps = kwargs["print_namespace_maps"]
```

Comment:
Take note that in #885 I also added typing to these keyword arguments so this will need to be added to the TypedDict definition added there.

Reply:
Added to `PrintSettings`

``` clojure
class PrintSettings(TypedDict, total=False):
    human_readable: bool
    print_dup: bool
    print_length: PrintCountSetting
    print_level: PrintCountSetting
    print_meta: bool
    print_namespace_maps: bool
    print_readably: bool
```